src/main/scala/tech/cryptonomic/conseil/Conseil.scala

```diff
@@ -7,61 +7,61 @@ import akka.http.scaladsl.model.StatusCodes._
-      }
-  }
+  applicationConfiguration match {
+    case Right((server, platforms, securityApi)) =>
 
-  val conf = ConfigFactory.load
-  val conseil_hostname = conf.getString("conseil.hostname")
-  val conseil_port = conf.getInt("conseil.port")
+      val validateApiKey = headerValueByName("apikey").tflatMap[Tuple1[String]] {
+        case Tuple1(apiKey) if securityApi.validateApiKey(apiKey) =>
+            provide(apiKey)
+        case _ =>
+            complete((Unauthorized, "Incorrect API key"))
+      }
 
-  implicit val system: ActorSystem = ActorSystem("conseil-system")
-  implicit val materializer: ActorMaterializer = ActorMaterializer()
-  implicit val executionContext: ExecutionContextExecutor = system.dispatcher
-  val sslConfig = AkkaSSLConfig()
+      implicit val system: ActorSystem = ActorSystem("conseil-system")
+      implicit val materializer: ActorMaterializer = ActorMaterializer()
+      implicit val executionContext: ExecutionContextExecutor = system.dispatcher
+      val sslConfig = AkkaSSLConfig() //Should get rid of this??
 
-  val route = cors() {
-    enableCORS {
-      validateApiKey { _ =>
-        logRequest("Conseil", Logging.DebugLevel) {
-          pathPrefix("tezos") {
-            Tezos(system.dispatchers.lookup("akka.tezos-dispatcher")).route
+      val route = cors() {
+        enableCORS {
+          validateApiKey { _ =>
+            logRequest("Conseil", Logging.DebugLevel) {
+              pathPrefix("tezos") {
+                Tezos(system.dispatchers.lookup("akka.tezos-dispatcher")).route
+              }
+            }
+          } ~ options {
+            // Support for CORS pre-flight checks.
+            complete("Supported methods : GET and POST.")
           }
         } ~ logRequest("Service route", Logging.DebugLevel) {
           pathPrefix("metadata") {
-            PlatformDiscovery(conf)(system.dispatchers.lookup("akka.tezos-dispatcher")).route
+            PlatformDiscovery(platforms)(system.dispatchers.lookup("akka.tezos-dispatcher")).route
           }
         }
-      } ~ options {
-        // Support for CORS pre-flight checks.
-        complete("Supported methods : GET and POST.")
       }
-    }
-  }
 
-  val bindingFuture = Http().bindAndHandle(route, conseil_hostname, conseil_port)
-  logger.info("Bonjour...")
+      val bindingFuture = Http().bindAndHandle(route, server.hostname, server.port)
+      logger.info("Bonjour...")
+
+      sys.addShutdownHook {
+        bindingFuture
+        .flatMap(_.unbind().andThen{ case _ => logger.info("Server stopped...")} )
+        .flatMap( _ => system.terminate())
+        .onComplete(_ => logger.info("We're done here, nothing else to see"))
+      }
 
-  sys.addShutdownHook {
-    bindingFuture
-    .flatMap(_.unbind().andThen{ case _ => logger.info("Server stopped...")} )
-    .flatMap( _ => system.terminate())
-    .onComplete(_ => logger.info("We're done here, nothing else to see"))
+    case Left(errors) =>
+      //nothing to do
```

Comment:
Should the error be outputted here rather than the config loader?

Reply:
I chose to concentrate the logging configuration concern in the configuration trait, such that the main app needs only to take appropriate measures, specific to the app.

This also simplifies consistent handling of logging messages for config errors, without exposing details in the main objects.

If you have a strong opinion against such choice, I'm open to changes.

Reply:
Sounds good.